pkg/ccl/crosscluster/logical/logical_replication_writer_processor.go

```diff
@@ -146,6 +149,7 @@ func newLogicalReplicationWriterProcessor(
 			StreamID:    streampb.StreamID(spec.StreamID),
 			ProcessorID: processorID,
 		},
+		dlqClient: &TableDeadLetterQueueClient{},
```

Comment:
nit: in the dlq file, i'd add.

```
func InitDeadLetterQueue() DeadLetterQueue {
  return &loggingDeadLetterQueue
}
``` 

That way, the row processor is totally naive to the implementation of the DLQ.